src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -34,8 +41,30 @@ public class EditCommandParser implements Parser<EditCommand> {
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
-                        PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_GRADE);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME_EDIT, PREFIX_PHONE_EDIT,
+                        PREFIX_EMAIL_EDIT, PREFIX_ADDRESS_EDIT, PREFIX_TAG_EDIT, PREFIX_GRADE_EDIT,
+                        OPTION_PRINT_NAME, OPTION_PRINT_PHONE, OPTION_PRINT_EMAIL,
+                        OPTION_PRINT_ADDRESS, OPTION_PRINT_TAG, OPTION_PRINT_GRADE);
+
+        if (argMultimap.getValue(OPTION_PRINT_NAME).isPresent()) {
```

Comment:
Hmm i think if its written like this then if the user enters
edit !n john !e some email
then it'll just ignore the email part and the isSingleFieldPresent method in EditCommand